Migrate from CircleCI to GH Actions for security scans

aliuk2012 opened the pull request:
### Related
- API - https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/3285
- CAS1 - https://github.com/ministryofjustice/hmpps-approved-premises-ui/pull/2442
- CAS2 - https://github.com/ministryofjustice/hmpps-community-accommodation-tier-2-ui/pull/904
- CAS3 - https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/pull/1263

Result of running
https://github.com/ministryofjustice/hmpps-github-actions/blob/main/docs/workflow-migration.md#automation

aliuk2012 committed:
Migrate from CircleCI to GH Actions for security scans

aliuk2012 committed:
Remove backup of circleci config file

froddd commented on .github/workflows/security_npm_dependency.yml:
```diff
@@ -0,0 +1,12 @@
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "22 9 * * MON-FRI" # Every weekday at 09:22 UTC
```
Could we manually edit these times so all the checks happen before 8am for all CASs? I can see why we might want them staggered (that would be a whole lot of alerts flooding Slack and other channels at once otherwise!) but it probably would be nice to be able to start the work day with all needed alerts ready to be consulted :)

aliuk2012 committed:
Update schedule time for scans to run so they are before business hours

The time was originally randomly selected as part of the migration scripts https://github.com/ministryofjustice/hmpps-github-actions/blob/20ef545b6466d2a6e8408985b640d4585b305839/migrate-typescript-security-jobs.bash#L25 and its better for us to know before the day start what vulnerabilities there are to look at

froddd reviewed (approved):
LGTM 👍
